### PR TITLE
feat: implement producer panel

### PR DESCRIPTION
**Pull request by ajborley**
- implements a producer panel, with doc and tests.

Contributes to: #59

Signed-off-by: Andrew Borley <borley@uk.ibm.com>

### CODE REVIEW

File: ui/src/Panels/Producer/Producer.view.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  const [messageValue, setMessageValue] = useState(
+    translate('MESSAGE_VALUE_DEFAULT', {}, true)
+  );
+  const debouncedSetMessageValue = debounce(setMessageValue, 100);
+
+  const onButtonClick = () => {
+    if (producerRunning) {
+      stop();
+    } else {
+      start({ value: messageValue });
```
Comment (matthew-chirgwin) [inline ui/src/Panels/Producer/Producer.view.js:47]: the key `value` needs to be `custom` as that is what the backend expects

File: ui/src/Panels/Producer/Producer.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    background-color: $ui-01;
+
+    &__count {
+        margin-top: 10%;
+    }
+    &__control {
+        display: flex;
+        justify-content: space-between;
+    }
+    &__Message {
```
Comment (matthew-chirgwin) [inline ui/src/Panels/Producer/Producer.scss:15]: uber picky, but could this be lower case?

File: ui/src/ReactCustomHooks/useTranslate/README.md [hunk 3]
```diff
@@ -113,14 +113,14 @@ will make up the corpus available for you to use. We recommend 'namespacing'
 your translation keys, eg `MYCOMPONENT_TEXT`, to reduce the chance of
 collisions.
 
-The full signiture of the returned function is as follows:
+The full signature of the returned function is as follows:
```
Comment (matthew-chirgwin) [inline ui/src/ReactCustomHooks/useTranslate/README.md:116]: Good spot!

File: ui/src/ReactCustomHooks/useTranslate/README.md [hunk 4]
```diff
@@ -113,14 +113,14 @@ will make up the corpus available for you to use. We recommend 'namespacing'
...
 
 `translate(key, inserts, returnAsString)`
 
 - `key` - string, required - the translation key to use to render the desired
 value
 - `inserts` - object, optional - a JS object where the keys match with the
-above specified `${}` delimeters. At run time, the value for this key is
+above specified `${}` delimiters. At run time, the value for this key is
```
Comment (matthew-chirgwin) [inline ui/src/ReactCustomHooks/useTranslate/README.md:123]: And here!

File: ui/src/Panels/Producer/Producer.feature [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    And responses are returned
+    Then it should display statistics about what has been produced
+
+Scenario: As a user, I am shown error states if production fails
+    Given I have a Producer panel
+    When I start the producer
+    And Error responses are returned
+    Then I should be shown error messages for those production failures
+
+Scenario: As a user, I can see what has been produced so far
```
Comment (matthew-chirgwin) [inline ui/src/Panels/Producer/Producer.feature:29]: And so I dont forget, a test will need to be added (ala the consumer) to cover the user selecting a message

File: ui/src/Panels/Producer/Producer.stories.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import { withInfo } from '@storybook/addon-info';
+import { Producer } from './index.js';
+import ProducerReadme from './README.md';
+
+import { producerMockWebsocket } from './Producer.assets.js';
+
+const renderHelper = () => () => {
+  const count = number('Count', 5);
+  return (
+    <Producer
```
Comment (matthew-chirgwin) [inline ui/src/Panels/Producer/Producer.stories.js:13]: You will also want to have this as well https://github.com/matthew-chirgwin/kafka-java-vertx-starter/blob/7a2230a4350ae3a61f0b83c87c4c58a75ba46873/ui/src/Panels/Consumer/Consumer.stories.js#L14 when integrating the selected message stuff

File: ui/src/Panels/Producer/Producer.view.js (hunk 1 above)
Comment (ajborley) [reply ui/src/Panels/Producer/Producer.view.js]: Thanks! Was convinced that wasn't correct but couldn't spot what it was supposed to be.

File: ui/src/Panels/Producer/Producer.scss (hunk 2 above)
Comment (ajborley) [reply ui/src/Panels/Producer/Producer.scss]: Done.

File: ui/src/Panels/Producer/Producer.feature (hunk 5 above)
Comment (ajborley) [reply ui/src/Panels/Producer/Producer.feature:35]: Added

File: ui/src/Panels/Producer/Producer.stories.js (hunk 6 above)
Comment (ajborley) [reply ui/src/Panels/Producer/Producer.stories.js]: Also added

**Pull request merged**